internal/ethapi/api.go

```diff
@@ -1091,6 +1091,8 @@ func newRevertError(result *core.ExecutionResult) *revertError {
 	err := errors.New("execution reverted")
 	if errUnpack == nil {
 		err = fmt.Errorf("execution reverted: %v", reason)
+	} else if len(result.Revert()) >= 4 {
+		err = fmt.Errorf("%#x", result.Revert())
```

Comment:
I don't quite understand this (and the other locations). The PR description says

>  we could consider directly returning the raw data information when the revert type is a custom type.

But in all these three places, you _only_ return the raw data if it's more than `4` bytes long. So if it's some custom 2-byte error code, it is not returned raw. What is the reason for this?